js/ui/codeView.js

```diff
@@ -1261,7 +1269,32 @@ var CodeViewManager = new Lang.Class({
     },
 
     _addSession: function(actor) {
-        this._sessions.push(new CodingSession({ app: actor }));
+        const session = new CodingSession({ app: actor });
+
+        // When the app is minimized the WM doesn't emit the destroy signal if
+        // the app is closed, for this reason we need to listen to the 'destroy'
+        // signal of the actor if it's minimized, to be able to remove from the
+        // CodingSession list and avoid possible gnome-shell crash
+        let destroyAppHandlerId = 0;
+        let destroyToolboxHandlerId = 0;
+        session.connect('minimized', (session) => {
```

Comment:
Inside this callback you’ll need to check whether session.app and session.toolbox exist before connecting the signals... especially the toolbox may not exist yet at this point.

Reply:
Sure, I've added the check here.